### PR TITLE
Feat/algo transaction builder factory

### PR DESCRIPTION
**Pull request by nimraarif21**
@ryan-bitgo @fdmirza @gibrangul95

### CODE REVIEW

File: modules/account-lib/src/coin/algo/transactionBuilder.ts [hunk 1]
```diff
@@ -309,6 +312,13 @@ export abstract class TransactionBuilder extends BaseTransactionBuilder {
     return this;
   }
 
+  setSigners(addrs: string | string[]): this {
+    const signers = addrs instanceof Array ? addrs : [addrs];
```
Comment (ryan-bitgo) [inline modules/account-lib/src/coin/algo/transactionBuilder.ts:316]: prefer `Array.isArray`

**Review by ryan-bitgo — CHANGES_REQUESTED** [review]
does this PR need to include the half signed txns changes? the diff is hard to read

File: modules/account-lib/src/coin/algo/transactionBuilder.ts (hunk 1 above)
Comment (nimraarif21) [reply modules/account-lib/src/coin/algo/transactionBuilder.ts]: I have changed this.

**Comment by nimraarif21** [comment]
> does this PR need to include the half signed txns changes? the diff is hard to read

My branch depends on the half signed txns changes for it to work correctly.

**Pull request closed without merging**